docs/language.md

```diff
@@ -6390,101 +6390,145 @@ transaction {
+    sendingAccount.balance > 0
+}
+```
 
-  ```
+If the `pre` block throws an error, or does not return `true` the remainder of the transaction is not executed and it will be completely reverted.
 
-- The **postcondition phase** (declared using the `post` keyword)
-  is where the transaction can check
-  that its functionality was executed correctly with specific condition checks.
+## **Execute**
 
-  If any of the condition checks result in `false`, the transaction will fail
-  and be completely reverted.
+The `execute` block does exactly what it says, it executes the main logic of the transaction. This block is optional, but it is a best practice to add your main transaction logic in the section, so it is explicit.
 
-  Only condition checks are allowed in this section. No actual computation
-  or modification of values is allowed.
+```swift
+execute {
+    // Invalid: Cannot access the authorized account object,
+    // as `account1` is not in scope
+    let resource <- account1.load<@Resource>(from: /storage/resource)
+    destroy resource
 
-  ```cadence,file=post.cdc
-    post {
-        result.balance == 30: "Balance after transaction is incorrect!"
-    }
+    // Valid: Can access any account's public Account object
+    let publicAccount = getAccount(0x03)
+}
+```
 
-  ```
+You **may not** access private `AuthAccount` objects in the `execute` block, but you may get an account's `PublicAccount` object, which allows reading and calling methods on objects that an account has published in the /public/ area of its account. (resources, contract methods, etc.)
 
-```cadence,file=transaction-declaration.cdc
-// Optional: Importing external types from other accounts using `import`.
-import HelloWorld from 0x01
+## **Post**
 
-transaction {
+Statements inside of the `post` block are used to verify that your transaction logic has been executed properly. It contains zero or more condition checks.
 
-    // Optional: type declarations and fields, which must be initialized in `prepare`.
+For example, the a transfer transaction might ensure that the final balance has a certain value, or e.g. it was incremented by a specific amount.
 
-    // The prepare phase needs to have as many account parameters
-    // as there are signers for the transaction.
-    //
+```swift
+post {
+    result.balance == 30: "Balance after transaction is incorrect!"
+}
+```
+
+If any of the condition checks result in `false`, the transaction will fail and be completely reverted.
+
+Only condition checks are allowed in this section. No actual computation or modification of values is allowed.
+
+**A Note about `pre` and `post` Blocks**
+
+Another function of the `pre` and `post` blocks is to help provide information about how the effects of a transaction on the accounts and resources involved. This is essential because users may want to verify what a transaction does before submitting it. `pre` and `post` blocks provide a way to introspect transactions before they are executed.
```

Comment:
```suggestion
Another function of the `pre` and `post` blocks is to help provide information about how the effects of a transaction on the accounts and resources involved. 
This is essential because users may want to verify what a transaction does before submitting it. 
`pre` and `post` blocks provide a way to introspect transactions before they are executed.
```